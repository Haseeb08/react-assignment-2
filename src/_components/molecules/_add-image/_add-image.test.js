import React from "react";
import { create } from "react-test-renderer";
import {
  render,
  screen,
  getByRole,
  getByPlaceholderText,
  fireEvent,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddImageBox from "./_add-image";

describe("Add Image box testing", () => {
  test("Matches the snapshot", () => {
    const input = create(<AddImageBox />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("onchange of url input box", () => {
    const mockOnChange = jest.fn();
    const { getByPlaceholderText } = render(
      <AddImageBox setimageUrl={mockOnChange} setimageTitle={mockOnChange} />
    );
    const input = getByPlaceholderText("Enter URL");
    fireEvent.change(input, { target: { value: "ab" } });
    fireEvent.change(getByPlaceholderText("Enter title"), { target: { value: "ab" } });
    expect(mockOnChange).toHaveBeenCalledTimes(2);
  });
});
