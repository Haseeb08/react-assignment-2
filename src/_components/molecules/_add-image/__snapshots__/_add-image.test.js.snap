// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Image box testing Matches the snapshot 1`] = `
<div
  className="makeStyles-addImageBox-1"
>
  <div
    className="makeStyles-flexItem-3"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3"
    >
      Enter image details
    </h3>
  </div>
  <div
    className="makeStyles-flexItem-3"
  >
    <div>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-5 MuiInputBase-fullWidth MuiInput-fullWidth"
        onClick={[Function]}
        required={true}
      >
        <input
          className="MuiInputBase-input MuiInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter URL"
          required={true}
          type="text"
        />
      </div>
    </div>
    <p
      className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
    />
  </div>
  <div
    className="makeStyles-flexItem-3"
  >
    <div>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-5 MuiInputBase-fullWidth MuiInput-fullWidth"
        onClick={[Function]}
        required={true}
      >
        <input
          className="MuiInputBase-input MuiInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter title"
          required={true}
          type="text"
        />
      </div>
    </div>
    <p
      className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-bttn-2 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="Add"
  >
    <span
      className="MuiButton-label"
    >
      Add
    </span>
  </button>
</div>
`;
